### PR TITLE
@zamojtel/rpg 116 equipment

### PR DESCRIPTION
**Pull request by zamojtel**
# Description

# Checklist

* [ ] Included code to test these changes
* [ ] Updated Jira

### CODE REVIEW

File: src/main/java/io/rpg/controller/Controller.java [hunk 1]
```diff
@@ -115,6 +117,7 @@ public void onKeyboardEvent(KeyboardEvent event) {
         case D -> currentModel.getPlayer().setRightPressed(true);
         case S -> currentModel.getPlayer().setDownPressed(true);
         case W -> currentModel.getPlayer().setUpPressed(true);
+        case E -> popupController.openInventoryPopup(currentModel.getPlayer().getInventory(),getWindowCenterX(),getWindowCenterY());
```
Comment (mhawryluk) [inline src/main/java/io/rpg/controller/Controller.java:120]:
```suggestion
        case E -> popupController.openInventoryPopup(currentModel.getPlayer().getInventory(), getWindowCenterX(), getWindowCenterY());
```

File: src/main/java/io/rpg/controller/PopupController.java [hunk 2]
```diff
@@ -48,6 +51,19 @@ public void openTextImagePopup(String text, Image image, int x, int y){
   public void openPointsPopup(int pointsCount, int x, int y) {
     openTextImagePopup("You earned " + pointsCount + " points!", coinImage, x, y);
   }
+  public void openInventoryPopup(Inventory inventory,int x, int y){
```
Comment (mhawryluk) [inline src/main/java/io/rpg/controller/PopupController.java:54]:
```suggestion
  
  public void openInventoryPopup(Inventory inventory,int x, int y){
```

File: src/main/java/io/rpg/view/InventoryGameObjectView.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import io.rpg.model.object.CollectibleGameObject;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+public class InventoryGameObjectView extends ImageView {
+    CollectibleGameObject collectibleGameObject;
+
+    public InventoryGameObjectView(CollectibleGameObject collectibleGameObject) {
+        this.collectibleGameObject=collectibleGameObject;
```
Comment (mhawryluk) [inline src/main/java/io/rpg/view/InventoryGameObjectView.java:11]:
```suggestion
        this.collectibleGameObject = collectibleGameObject;
```

**Review by mhawryluk — COMMENTED** [review]
Great!

File: src/main/java/io/rpg/controller/Controller.java [hunk 4]
```diff
@@ -148,8 +151,12 @@ public void onMouseClickedEvent(MouseClickedEvent event) {
       if (object instanceof CollectibleGameObject) {
         popupController.openTextImagePopup("Picked up an item!", objectView.getImage(), getWindowCenterX(), getWindowCenterY());
         objectView.setVisible(false);
+        currentModel.getPlayer().getInventory().add((CollectibleGameObject) object);
```
Comment (mhawryluk) [inline src/main/java/io/rpg/controller/Controller.java:154]: replace that with an Action implementation

**Review by kkafar — APPROVED** [review]
It works, but it is still in development stage. Do not merge it to master yet.

**Pull request merged**